build.gradle

```diff
@@ -1585,6 +1644,7 @@ project(':shell') {
     implementation libs.jacksonJDK8Datatypes
     implementation libs.jline
     implementation libs.slf4jApi
+    implementation project(':server-common')
```

Comment:
Right, this is needed because `core` module does not expose `server-common` as a transitive module with gradle `api` dependency. It uses `implementation` dependency for `server-common` to be part of `core` module's compile/runtime classpaths.

I am not sure whether we really have usages for `server-common` classes to be exposed as part of `core` module.

Reply:
core should generally not expose anything with api since core doesn't expose any public api.